tsdb/engine/tsm1/compact.gen.go

```diff
@@ -439,6 +439,220 @@ func (k *tsmKeyIterator) chunkInteger(dst blocks) blocks {
 }
 
 // merge combines the next set of blocks into merged blocks.
+func (k *tsmKeyIterator) mergeUnsigned() {
+	// No blocks left, or pending merged values, we're done
+	if len(k.blocks) == 0 && len(k.merged) == 0 && len(k.mergedUnsignedValues) == 0 {
+		return
+	}
+
+	dedup := len(k.mergedUnsignedValues) != 0
+	if len(k.blocks) > 0 && !dedup {
+		// If we have more than one block or any partially tombstoned blocks, we many need to dedup
+		dedup = len(k.blocks[0].tombstones) > 0 || k.blocks[0].partiallyRead()
+
+		// Quickly scan each block to see if any overlap with the prior block, if they overlap then
+		// we need to dedup as there may be duplicate points now
+		for i := 1; !dedup && i < len(k.blocks); i++ {
```

Comment:
Good call – the resulting ASM is simpler too. I'll update the code-gen templates.